pkg/tsdb/intervalv2/intervalv2.go

```diff
@@ -82,31 +82,40 @@ func (ic *intervalCalculator) CalculateSafeInterval(timerange backend.TimeRange,
 // queryIntervalMS is a pre-calculated numeric representation of the query interval in milliseconds.
 func GetIntervalFrom(dsInterval, queryInterval string, queryIntervalMS int64, defaultInterval time.Duration) (time.Duration, error) {
 	// Apparently we are setting default value of queryInterval to 0s now
-	if queryInterval == "0s" {
-		queryInterval = ""
+	interval := queryInterval
+	if interval == "0s" {
+		interval = ""
 	}
-
-	if queryInterval == "" {
+	if interval == "" {
 		if queryIntervalMS != 0 {
 			return time.Duration(queryIntervalMS) * time.Millisecond, nil
 		}
 	}
-	interval := queryInterval
-	if queryInterval == "" && dsInterval != "" {
+	if interval == "" && dsInterval != "" {
 		interval = dsInterval
 	}
 	if interval == "" {
 		return defaultInterval, nil
 	}
-	interval = strings.Replace(strings.Replace(interval, "<", "", 1), ">", "", 1)
-	isPureNum, err := regexp.MatchString(`^\d+$`, interval)
+
+	parsedInterval, err := ParseIntervalStringToTimeDuration(interval)
+	if err != nil {
+		return time.Duration(0), err
+	}
+
+	return parsedInterval, nil
+}
+
+func ParseIntervalStringToTimeDuration(interval string) (time.Duration, error) {
```

Comment:
I extracted this too separate method as I needed this part in `calculateRateInterval`